clearpath_config/platform/attachments/a200.py

```diff
@@ -26,171 +26,88 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 # A200 Husky Platform Configuration
-from clearpath_config.common.types.config import BaseConfig
+from clearpath_config.common.types.accessory import Accessory
 from clearpath_config.common.types.platform import Platform
-from clearpath_config.common.utils.dictionary import flip_dict
-from clearpath_config.platform.attachments.config import BaseAttachmentsConfig
+from clearpath_config.platform.types.attachment import BaseAttachment, PlatformAttachment
 from clearpath_config.platform.types.bumper import Bumper
-from clearpath_config.platform.types.top_plate import TopPlate
-from clearpath_config.platform.types.structure import Structure
+from typing import List
 
 
-# A200 Husky Attachments Configuration
-class A200AttachmentsConfig(BaseConfig, BaseAttachmentsConfig):
+class A200TopPlate(BaseAttachment):
     PLATFORM = Platform.A200
+    ATTACHMENT_MODEL = "%s.top_plate" % PLATFORM
+    DEFAULT = "default"
+    LARGE = "large"
+    PACS = "pacs"
+    MODELS = [DEFAULT, LARGE, PACS]
+    PARENT = "default_mount"
 
-    ATTACHMENTS = "attachments"
-    FRONT_BUMPER = "front_bumper"
-    REAR_BUMPER = "rear_bumper"
-    TOP_PLATE = "top_plate"
-    STRUCTURE = "structure"
-
-    TEMPLATE = {
-        ATTACHMENTS: {
-            FRONT_BUMPER: FRONT_BUMPER,
-            REAR_BUMPER: REAR_BUMPER,
-            TOP_PLATE: TOP_PLATE,
-            STRUCTURE: STRUCTURE,
-        }
-    }
+    def __init__(
+            self,
+            name: str = ATTACHMENT_MODEL,
+            model: str = DEFAULT,
+            enabled: bool = BaseAttachment.ENABLED,
+            parent: str = PARENT,
+            xyz: List[float] = Accessory.XYZ,
+            rpy: List[float] = Accessory.RPY
+            ) -> None:
+        super().__init__(name, model, enabled, parent, xyz, rpy)
 
-    KEYS = flip_dict(TEMPLATE)
 
-    DEFAULTS = {
-        FRONT_BUMPER: {
-            'name': FRONT_BUMPER,
-            'enabled': Bumper.ENABLED,
-            'extension': Bumper.EXTENSION,
-            'model': Bumper.DEFAULT},
-        REAR_BUMPER: {
-            'name': REAR_BUMPER,
-            'enabled': Bumper.ENABLED,
-            'extension': Bumper.EXTENSION,
-            'model': Bumper.DEFAULT
-            },
-        TOP_PLATE: {
-            'name': TOP_PLATE,
-            'enabled': TopPlate.ENABLED,
-            'model': TopPlate.DEFAULT
-            },
-        STRUCTURE: {
-            'name': STRUCTURE,
-            'enabled': Structure.ENABLED,
-            'model': Structure.DEFAULT
-            },
-    }
+class A200Bumper(Bumper):
+    PLATFORM = Platform.A200
+    ATTACHMENT_MODEL = "%s.bumper" % PLATFORM
+    PLATFORM = Platform.A200
```

Comment:
Duplicate

Reply:
This is not a duplicate. It is the bumper of the A200, when we add bumpers to other platforms they'll have their own `PLATFORMBumper` class.

Reply:
I meant the line `PLATFORM = Platform.A200` is written twice here

Reply:
Good catch, thanks